hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/exception/EventLimitExceededException.java

```diff
@@ -0,0 +1,16 @@
+package com.hazelcast.jet.impl.exception;
+
+import com.hazelcast.jet.JetException;
+import com.hazelcast.jet.impl.TerminationMode;
+
+import javax.annotation.Nonnull;
+
+public class EventLimitExceededException extends JetException {
+    public EventLimitExceededException() {
+        this(TerminationMode.CANCEL_GRACEFUL);
```

Comment:
`CANCEL_GRACEFUL` is cancellation with a snapshot, an enterprise feature. 
And also what's the reason we're using this enum here? Just to get the text?

Reply:
I also don't like the idea of controlling the flow with an exception.